cfme/tests/candu/test_vm_graph.py

```diff
@@ -81,11 +85,39 @@ def refresh():
     assert graph_data > 0
 
 
-# To-Do add support for other provider
-@pytest.mark.provider([VMwareProvider], scope='module', override=True)
+def compare_data(tb_data, gp_data, legends, tolerance=1):
+    """ Compare Utilization graph and table data.
+    Args:
+        tb_data : Data from Utilization table
+        gp_data : Data from Utilization graph
+        legends : Legends in graph; which will help for comparison
+        tolerance : Its error which we have to allow while comparison
+    """
+    for row in tb_data:
+        for key, data in gp_data.items():
+            if any([re.match(key, item) for item in row['Date/Time'].split()]):
+                for leg in legends:
+                    tb = row[leg].replace(',', '').replace('%', '').split()
+                    if tb:
+                        tb = round(float(tb[0]), 1)
+                        gp = round(
+                            float(data[leg].replace(',', '').replace('%', '').split()[0]), 1)
+                        assert abs(tb - gp) <= tolerance
+                    else:
+                        logger.warning("No {leg} data captured for DateTime: {dt}".format(
+                            leg=leg, dt=row['Date/Time']))
+
+
+@pytest.mark.uncollectif(lambda provider, interval, graph_type:
```

Comment:
Required: You can override provider here and specify what you need
ex. @pytest.mark.provider([VMwareProvider], override=True)

Reply:
@rlbabyuk: It's not possible with provider overriding. It can be used only when the uncollect decision depends only on the `provider`. But, In this case, we are uncollecting things with `provider`, `Interval` and `graph type`.